packages/dev/core/src/Materials/shaderMaterial.ts

```diff
@@ -596,6 +596,32 @@ export class ShaderMaterial extends PushMaterial {
         return this;
     }
 
+    /**
+     * Adds or removes the specified shader define. Note if the active defines do change,
+     * the shader will be recompiled and this can be expensive.
+     * @param define the define string e.g., "OUTPUT_TO_SRGB"
+     * @param isEnabled true if the define should be enabled, false if it should be disabled
+     * @returns the material itself allowing "fluent" like uniform updates
+     */
+    public setDefine(define: string, isEnabled: boolean): ShaderMaterial {

```

Comment:
I think the implementation needs to be a bit more general.

A define in `options.defines` may or may not start with `#define`, so you should take this into account.

Also, we handle defines that have values, not just boolean defines. I think the simplest way to handle this is to allow either a boolean or a string to be passed:

```typescript
public setDefine(define: string, value: boolean | string): ShaderMaterial {
```
and take this into account in the implementation.

Reply:
The options for ShaderMaterial only support strings for the define (and not a name plus bool value), and I think it's fair to stay in sync with that here.

I could clarify in the function doc but I was expecting the caller would be matching the define string they passed into the options, which may or may not include the prefixed `#define` string. If you'd still prefer the implementation test both combinations (the string with `#define ` prefixed and one without it), I can do it that way too. I'm just not sure it's worth it. I figured it would be strange for a caller to pass `\"#define MY_DEFINE\"` through the options but then decide to call this function with just `"MY_DEFINE"` (or vice versa).

Reply:
I agree we can expect the user to use `#define` in the string if they used it in the first place.

However, you can pass a define like `#define MYVALUE 23` in the options, you are not limited to boolean defines.

So, we could call `setDefines("#define MYVALUE", "24")` to change the value.

Reply:
@kircher1 and @Popov72 let me know when ready so that I can merge it.

Reply:
Hi @Popov72  and @sebavan , I am taking a look now at the latest feedback. Will try and add a value field.

Reply:
I updated the logic to handle a value. Let me know if that works or if it needs another pass! Thanks!